### PR TITLE
fix: estimateMessageFee - eth address format

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution
Solve #1028 
For an Ethereum address, the Starknet rpc specs 0.6.0 and 0.7.0 are waiting for  :
(https://github.com/starkware-libs/starknet-specs/blob/4233164155b8a279b408b9638c34db2602b5cdc0/api/starknet_api_openrpc.json#L1322-L1328)
The hex string needs exactly 20 bytes lenght.
Today, length is not formatted to this specific value, and generate an error in `estimateMessageFee`. 
Corrected in both rpc versions.

## Usage related changes

None.

## Development related changes

- A new function has been added in utils/eth.ts :
```typescript
function validateAndParseEthAddress(address: BigNumberish): string
```

- In tests/rpcprovider, a Cairo 1 contract test as been added, using a non 20 bytes length address, and it works properly.
The current Cairo 0 contract test is still performed with a complete 20 bytes length, because the contract is verifying that the right value is passed in the parameter; So, a non 20 bytes length address can't be tested with this contract.

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: __tests__/utils/ethSigner.test.ts [hunk 1]
```diff
@@ -321,4 +322,18 @@ describe('Ethereum signer', () => {
       );
     });
   });
+  describe('Ethereum address', () => {
```
Comment (ivpavici) [inline __tests__/utils/ethSigner.test.ts:325]: can we add a test for the `Invalid Ethereum Address Format` case as well?

**Review by tabaktoni — APPROVED** [review]
mentioned, base to next-version and can be merged

File: __tests__/utils/ethSigner.test.ts (hunk 1 above)
Comment (PhilippeR26) [reply __tests__/utils/ethSigner.test.ts:325]: It's already the case (one byte missing).
I added a comment to clarify.

**Comment by PhilippeR26** [comment]
All done.

**Comment by SergeyVolynkin** [comment]
Hi @PhilippeR26 @tabaktoni, thank you for this fix! Would it be possible to cherry-pick this fix into `5.24.3` stable version? We have one of our contracts starting as `0x0…` which breaks some funds deposit calls on `5.24.3` (we're on starknet appchain)

**Comment by PhilippeR26** [comment]
You can consider the v6.6.6 as stable ; just the new Wallet Account is not finalized.

**Comment by SergeyVolynkin** [comment]
@PhilippeR26 I believe this PR is only included in `6.7.0`? 
Given `6.7.0` is not yet stable, would you consider cherry-picking'ing this fix to the `5.24.*` version?

**Comment by SergeyVolynkin** [comment]
Hi @ivpavici, the `1-2 weeks` timeline for `v6` release sounds good -- I've triaged the migration changes needed and will be able to update our app with relative ease.

Will be waiting for the `v6` release then, thank you @ivpavici!

**Comment by SergeyVolynkin** [comment]
Hi @ivpavici, sorry, following up if you know the new draft release date for sta stable v6?

**Comment by ivpavici** [comment]
@SergeyVolynkin hi!

We are a bit blocked by wallets, they have to release a stable version for the new wallet communication as per this [snip](https://community.starknet.io/t/snip-standardizing-starknet-wallet-connections-with-write-api-and-json-rpc-integration/98049)

We have been promised to get the wallets updated and it got delayed by over the last weeks. Tomorrow we should receive the latest Braavos verison, but Argent not before next week I'm afraid

But apart from this, we consider the latest pre-releases as stable

**Comment by SergeyVolynkin** [comment]
@ivpavici thank you very much for the update

**Pull request merged**